app/Http/Controllers/Api/RequestApiController.php

```diff
@@ -31,15 +28,24 @@ public function indexAllRequests(?string $format = null)
      * Find request in the database.
      * Throw 404 when request is not found.
      *
-     * @param  int  $requestId
-     * @param  null|string  $format
-     * @return Response
+     * @param int $requestId
+     * @param null|string $format
+     * @return \App\Http\Resources\Html\Request|\App\Http\Resources\Json\Request|\App\Http\Resources\Xml\Request
      */
     public function showSingleRequest(int $requestId, ?string $format = null)
     {
-        $request = \App\Request::findOrFail($requestId);
+        $request = EndpointRequest::with('endpoint')
+            ->with('profile')
```

Comment:
You can do `->with('profile', 'requestHeaders', 'responses', 'responses.responseHeaders')->...` 🤓

Reply:
Even better:
->with([
    'profile',
    'requestHeaders',
    ... 
])